uk.co.compucorp.civicrm.hrleaveandabsences/templates/CRM/HRLeaveAndAbsences/Form/AbsenceType.tpl

```diff
@@ -128,154 +132,200 @@
+                    {/literal}
+                    {else}
+                    {literal}
+                    CRM.alert("This leave/absence type is in use and cannot be deleted. Please disable it instead.",
+                      'Delete Leave/Absence type', 'error');
+                    {/literal}
+                    {/if}
+                    {literal}
 
-                  function hideCarryForwardExpirationDuration() {
-                      document.getElementById('carry_forward_expiration_duration').value = '';
-                      $('#carry_forward_expiration_unit').select2('val', '');
-                      $('.carry-forward-expiration-duration').hide();
-                  }
+                  });
+                }
 
-                  function initColorPicker() {
-                      $('#color').spectrum({
-                          preferredFormat: "hex3",
-                          allowEmpty:true,
-                          showPaletteOnly: true,
-                          showPalette:true,
-                          {/literal}
-                          palette: {$availableColors}
-                          {literal}
-                      });
-                  }
+                function deleteCallback() {
+                  {/literal}
+                  window.location = "{$deleteUrl}";
+                  {literal}
+                }
+
+                function initCalculationUnitControls() {
+                  calculationUnitRelatedControls();
 
-                  function initDeleteButton() {
-                      $('.crm-button-type-delete').on('click', function(e) {
-                        e.preventDefault();
-                        {/literal}
-                        {if $canDeleteType}
-                        {literal}
-                            CRM.confirm({
-                              title: ts('Delete Leave/Absence type'),
-                              message: ts('Are you sure you want to delete this leave/absence type?'),
-                              options: {
-                                yes: ts('Yes'),
-                                no: ts('No')
-                              }
-                            })
-                            .on('crmConfirm:yes', deleteCallback);
-                        {/literal}
-                        {else}
-                        {literal}
-                            CRM.alert("This leave/absence type is in use and cannot be deleted. Please disable it instead.",
-                                      'Delete Leave/Absence type', 'error');
-                        {/literal}
-                        {/if}
-                        {literal}
+                  $('.absence-calculation-unit select').on('change', function(e) {
+                    calculationUnitRelatedControls();
+                  });
+                }
 
-                      });
+                function calculationUnitRelatedControls() {
```

Comment:
Since what this method does is just toggling some fields, how about renaming it to `toggleCalculationUnitRelatedControls`?